### PR TITLE
Backend api refactor

### PR DESCRIPTION
**Pull request by hey-waffles**

### CODE REVIEW

File: src/server/apiConfigBuilder/ApiConfigBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
+
```
Comment (hey-waffles) [inline src/server/apiConfigBuilder/ApiConfigBuilder.ts:1]: Remove top line

File: src/server/apiConfigBuilder/ApiConfigBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  }
+
+  /**
+   * Indicates the end of the logic building. Returns a struct containing functions for accessing fauna
+   */
+  public done() {
+    return this;
+  }
+
+  public export() {
```
Comment (hey-waffles) [inline src/server/apiConfigBuilder/ApiConfigBuilder.ts:37]: Comment

File: src/server/apiConfigBuilder/ApiConfigBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    return this.config;
+  }
+
+  /**
+   * Manually sets a configuration field with function information
+   * @param name The name of the configuration field to set
+   * @param config The function configuration information
+   */
+  public set(name: string, config: FunctionConfig) {
```
Comment (hey-waffles) [inline src/server/apiConfigBuilder/ApiConfigBuilder.ts:46]: Type these functions

File: src/server/apiConfigBuilder/ApiConfigBuilder.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    return new SearchBuilder(name, indexName, this);
+  }
+
+  private checkIfCanSetDefault() {
+    if (this.canSetDefaults === false) {
+      throw "You can no longer set the default values. Move any default setter to the top of the configuration class";
+    }
+  }
+
+  public login(canLogin: boolean) {
```
Comment (hey-waffles) [inline src/server/apiConfigBuilder/ApiConfigBuilder.ts:102]: Comment functions

File: src/server/apiConfigBuilder/classes/FieldBuilder.ts [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+      } else {
+        value = this.config[i];
+      }
+    }
+
+    this.parent.set(this.name, this.config);
+    return this.parent;
+  }
+
+  public static default(): RoleConfig {
```
Comment (hey-waffles) [inline src/server/apiConfigBuilder/classes/FieldBuilder.ts:44]: Comment

File: src/server/apiConfigBuilder/classes/SearchBuilder.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import { Expr } from "faunadb";
+import { MyUserDocument } from "types/security";
+import { ApiConfigBuilder } from "../ApiConfigBuilder";
+import { FunctionType } from "../types";
+import { FunctionBuilder } from "./FunctionBuilder";
+
+export class SearchBuilder extends FunctionBuilder {
```
Comment (hey-waffles) [inline src/server/apiConfigBuilder/classes/SearchBuilder.ts:7]: Comment

File: src/server/apiConfigBuilder/methods/fetch.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
+
```
Comment (hey-waffles) [inline src/server/apiConfigBuilder/methods/fetch.ts:1]: Remove top line

File: src/server/auth.ts [hunk 8]
```diff
@@ -1,31 +1,62 @@
...
 export function getMyUser(req: NextApiRequest): MyUserDocument {
   // TODO - have this working on page refresh
- const myUser = getSession({req});
- if (myUser === null || !("user" in myUser)) {
-  const myUser2: MyUserDocument = {
-    id: "295863299256353286",
-    collection: "users",
-    ref: toFaunaRef({
+  const myUser = getSession({req});
+  console.log(myUser)
```
Comment (hey-waffles) [inline src/server/auth.ts:9]: No console.logs

File: src/types/security/index.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import { Expr } from "faunadb";
+import { FaunaRef } from "types/fauna";
+
+export enum UserRole {
```
Comment (hey-waffles) [inline src/types/security/index.ts:4]: Comment

**Review by lucyawrey — APPROVED** [review]
No issues, tested a build manually on my local machine due to having issues with CI/CD.

**Pull request merged**